src/Tools/dotnet-monitor/StartupHook/StartupHookValidator.cs

```diff
@@ -48,6 +48,21 @@ public async Task<bool> CheckAsync(IEndpointInfo endpointInfo, CancellationToken
             }
 
             DiagnosticsClient client = new(endpointInfo.Endpoint);
+
+            if (endpointInfo.RuntimeVersion.Major >= 8)
+            {
+                try
+                {
+                    await client.ApplyStartupHookAsync(startupHookLibraryFileInfo.PhysicalPath, token);
+
+                    return true;
+                }
+                catch (Exception ex)
+                {
+                    _logger.StartupHookApplyFailed(ex);
```

Comment:
Currently, this logs that the automatic apply was unsuccessful before we check the environment variables - do we want this behavior, or do we only want to log if the env var also wasn't provided?